crates/ide/src/references.rs

```diff
@@ -309,7 +309,10 @@ fn handle_control_flow_keywords(
     let file = sema.parse_guess_edition(file_id);
     let edition =
         sema.attach_first_edition(file_id).map(|it| it.edition()).unwrap_or(Edition::CURRENT);
-    let token = file.syntax().token_at_offset(offset).find(|t| t.kind().is_keyword(edition))?;
+    let token = file
+        .syntax()
+        .token_at_offset(offset)
+        .find(|t| t.kind().is_keyword(edition) || t.kind() == T![=>])?;
```

Comment:
Lets use the pick function here https://github.com/rust-lang/rust-analyzer/blob/aded72f8e6048b243e20e0b77a3870b1b76ef62d/crates/ide-db/src/helpers.rs#L28